### PR TITLE
NIAD_1693 Map Clinical Document Effective Time

### PR DESCRIPTION
**Pull request by ssokolowskisebastian**

### CODE REVIEW

File: service/src/integration-test/java/uk/nhs/adaptors/oneoneone/cda/report/controller/ReportControllerIT.java [hunk 1]
```diff
@@ -83,6 +83,7 @@
         "entry[*].resource.data[*].reference.reference",
         "entry[*].resource.date",
         "entry[*].resource.dateTime",
+        "entry[*].resource.destination[0].endpoint",
```
Comment (mrogowska) [inline service/src/integration-test/java/uk/nhs/adaptors/oneoneone/cda/report/controller/ReportControllerIT.java:86]: Why is this ignored?

File: service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/service/EncounterReportService.java [hunk 2]
```diff
@@ -30,7 +32,7 @@
     private final AmqpProperties amqpProperties;
 
     public void transformAndPopulateToGP(POCDMT000002UK01ClinicalDocument1 clinicalDocumentDocument,
-        String messageId, ItkReportHeader header) throws XmlException {
+        String messageId, ItkReportHeader header) throws XmlException, ParseException {
```
Comment (mrogowska) [inline service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/service/EncounterReportService.java:35]: `ParseException` is not thrown here

File: service/src/test/java/uk/nhs/adaptors/oneoneone/cda/report/service/EncounterReportServiceTest.java [hunk 3]
```diff
@@ -63,7 +65,7 @@ public void setUp() {
     }
 
     @Test
-    public void shouldTransformAndPopulateToGP() throws JMSException, XmlException {
+    public void shouldTransformAndPopulateToGP() throws JMSException, XmlException, ParseException {
```
Comment (mrogowska) [inline service/src/test/java/uk/nhs/adaptors/oneoneone/cda/report/service/EncounterReportServiceTest.java:68]: `ParseException` is not thrown here

File: service/src/test/java/uk/nhs/adaptors/oneoneone/cda/report/controller/ReportControllerTest.java [hunk 4]
```diff
@@ -66,7 +67,7 @@
     private ItkAddressValidator itkAddressValidator;
 
     @Test
-    public void postReportValidRequest() throws XmlException {
+    public void postReportValidRequest() throws XmlException, ParseException {
```
Comment (mrogowska) [inline service/src/test/java/uk/nhs/adaptors/oneoneone/cda/report/controller/ReportControllerTest.java:70]: `ParseException` is not thrown here

File: service/src/test/java/uk/nhs/adaptors/oneoneone/cda/report/service/EncounterReportBundleServiceTest.java [hunk 5]
```diff
@@ -272,7 +279,7 @@ public void setUp() throws XmlException {
 
     @Test
     @SuppressWarnings("MagicNumber")
-    public void shouldCreateEncounterBundle() throws XmlException {
+    public void shouldCreateEncounterBundle() throws XmlException, ParseException {
```
Comment (mrogowska) [inline service/src/test/java/uk/nhs/adaptors/oneoneone/cda/report/service/EncounterReportBundleServiceTest.java:282]: `ParseException` is not thrown here

File: service/src/integration-test/resources/json/expectedResult.json [hunk 6]
```diff
@@ -7,3785 +7,2820 @@
     "value": "2"
   },
   "type": "message",
-  "entry": [
```
Comment (mrogowska) [inline service/src/integration-test/resources/json/expectedResult.json:10]: Can you please go back to previous style of braces and brackets? It's really hard to see what really changed here. The same with expectedResult2.json

**Review by mrogowska — CHANGES_REQUESTED** [review]

**Review by dawidskainos — CHANGES_REQUESTED** [review]

File: service/src/integration-test/resources/json/expectedResult.json [hunk 7]
```diff
@@ -20,10 +20,10 @@
         },
         "destination": [
           {
-            "endpoint": "urn:nhs-uk:addressing:ods:EM396"
+            "endpoint": "urn:nhs-uk:addressing:ods:EM396:DOSServiceID:26428"
```
Comment (mrogowska) [inline service/src/integration-test/resources/json/expectedResult.json:23]: I think only expectedResult2 should expect this endpoint value (only ITK_Report_request_2.xml has both ODS Code and DOS Service ID).

File: service/src/test/java/uk/nhs/adaptors/oneoneone/cda/report/service/MessageHeaderServiceTest.java [hunk 8]
```diff
@@ -40,6 +42,7 @@
     @BeforeEach
     public void setUp() {
         when(soapProperties.getSendTo()).thenReturn(ENDPOINT);
+
```
Comment (mrogowska) [inline service/src/test/java/uk/nhs/adaptors/oneoneone/cda/report/service/MessageHeaderServiceTest.java:45]: Unnecessary empty line

**Review by mrogowska — CHANGES_REQUESTED** [review]

**Review by mrogowska — APPROVED** [review]

**Review by mrogowska — APPROVED** [review]

**Review by dawidskainos — APPROVED** [review]

**Pull request merged**